src/triplot.jl

```diff
@@ -0,0 +1,30 @@
+Makie.plottype(::Triangulation) = Viz{<:Tuple{SimpleMesh}}
+
+function _convert(::Type{SimpleMesh}, tri::Triangulation) # for plotting, we can drop some information
+    # Be a bit explicit, in case of ghost triangles. Don't need to use each_solid_vertex 
+    # here, any missing vertices just won't get plotted as they won't appear in any 
+    # connectivities. Using it messes up the order anyway (each_solid_vertex accesses in a 
+    # different order than 1, 2, ….)
+    points = Meshes.Point2[]
+    triangles = NTuple{3,Int64}[]
+    for p in DelaunayTriangulation.each_point(tri)
+        push!(points, Meshes.Point2(p))
+    end
+    for τ in DelaunayTriangulation.each_solid_triangle(tri)
+        ijk = DelaunayTriangulation.indices(τ)
+        push!(triangles, ijk)
+    end
+    triangles = connect.(triangles, Triangle)
+    return SimpleMesh(points, triangles)
+end
```

Comment:
I do think that `SimpleMesh` is the closest thing to `Triangulation`, although 
they are not exactly the same. `SimpleMesh` seems to be a point set equipped
with some topology, while `Triangulation` is supposed to represent 
a collection of points, edges, and polygons (i.e. a piecewise linear complex).

In terms of what I need from Triangulation that I can't get from SimpleMesh, 
I need ways to distinguish edges from just constrained edges so that 
a piecewise linear complex isn't treated the same as a point set, or refining a 
mesh won't push past a constrained edge. I also need ways of enumerating the boundary 
so that I know (1) what curve I'm on, and (2) what segment of the curve I am on.
I don't think I can get that from SimpleMesh? Also, ghost triangles (edges on the 
boundary that adjoining a point at infinity, with different points for different 
contiguous boundaries or different chains on a contiguous boundary - based on 
the definition of HalfEdge, I think you only allow for Nothing to represent 
this vertex?) are extremely important.

The complete `Triangulation` data structure is (it's changed since the last time 
we talked - I don't think it should change again):

```julia
struct Triangulation{P,Ts,I,E,Es,BN,B,BIR}
    points::P
    triangles::Ts
    adjacent::Adjacent{I,E}
    adjacent2vertex::Adjacent2Vertex{I,Es,E}
    graph::Graph{I}
    boundary_nodes::BN
    boundary_map::B
    boundary_index_ranges::BIR
    constrained_edges::Es
    all_constrained_edges::Es
    convex_hull::ConvexHull{P,Vector{I}}
end
```

This is similar to `SimpleMesh` if I were to write it as something like 

```julia
struct Triangulation{P, Es, BN, T}
    points::P 
    constrained_edges::Es 
    boundary_nodes::BN 
    topology::T
end 
```

with `topology` containing everything else. These fields are all pretty important,
especially for later applications like mesh refinement where looking up where we are on 
a boundary, or all neighbours, etc. needs to be O(1) so that things like point location, 
segment location, etc. are fast, especially mutating them as new triangles, edges, and points are added or deleted. 

Note that the `convex_hull` field is important for 
unconstrained triangulations since it stores the boundary (and I might need `convex_hull`
later for restoring the Delaunay property of perturbed meshes, when I get to it). 

The 
`boundary_map` field is needed so that I can map from a boundary index (these are numbers 
`-1`, `2`, and so on for enumerating different parts of a boundary) to its position 
in `boundary_map`. `boundary_index_ranges` maps a boundary index to all other boundary indices 
that appear on a contiguous curve (e.g. a square with four distinct sides has boundary indices 
`-1`, `-2`, `-3`, and `-4`, and a square with a hole would have those four boundary indices on the 
square and `-5` on the hole).) 

I'll also note that most of what I read is best understood in terms of these adjacent/adjacent2vertex, etc. 
maps, so that's another reason I want to keep it. Some of these things get a bit complicated, so I don't want to increase the mental load so much in developing.